ArchitectureFlow/ArchitectureFlow/Login/Presentation/UI/LoginViewController.m

```diff
@@ -0,0 +1,47 @@
+#import "LoginViewController.h"
+#import "LoginPresenter.h"
+
+@interface LoginViewController ()
+
+@property (nonatomic) LoginPresenter *presenter;
+@property (weak, nonatomic) IBOutlet UITextField *emailTextField;
+@property (weak, nonatomic) IBOutlet UITextField *passwordTextField;
+@property (weak, nonatomic) IBOutlet UISwitch *stayLoggedInSwitch;
+
+@end
+
+@implementation LoginViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    // Do any additional setup after loading the view, typically from a nib.
+}
+
+- (IBAction)loginButtonTapped:(id)sender {
+    
+    [self.presenter loginTapped];
+}
+
+#pragma mark - Login View Model
```

Comment:
It's pretty interesting that we implement the viewmodel with the VC instead of using a pure immutable data class. I see some of its advantages (eg there is 1 single source of truth), but I'm not sure it works out that well for more complex examples, so it would be good to discuss pros/cons.

Reply:
Again I've talked about this in my presentation. This is because our UI typically IS the view Model, we just don't think about it this way. This is just an example making it clear, but it's not to say we should have separate interfaces, we can have just one View interface if that is more convinient (it'll just d 2 jobs. We can merge the View and ViewModel interfaces because again their are no-ops), but we do need at least 1, because our presenter cannot know about the actual ViewController. But back to UI being the view model.
So say we have a login screen, then the view model for the screen is:
the username, the password etc. It's the state the view is in currently. So the easiest way to access the view model is to make the view implement that protocol by itself because it can just ask it's views for the text etc, but we could have a data class kept in sync too. The common misconception is that the data passed to the view from the presenter is ViewModel, which not quite correct, it might be a part of view model however.